src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -114,6 +114,11 @@ public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType p
             _socketType = socketType;
             _protocolType = protocolType;
 
+            if (addressFamily == AddressFamily.InterNetworkV6)
+            {
+                DualMode = false;
```

Comment:
I'm concerned that changing to `Debug.Assert()` might impact execution in debug mode since it makes a call to a Win32 API. Is there any kind of in-memory state that we can probe without calling an API?